crates/sqlexec/src/remote/rewriter.rs

```diff
@@ -72,8 +75,14 @@ impl TreeNodeRewriter for DDLRewriter {
             match ext_type {
                 ExtensionType::Insert => {
                     let insert_into = ext.node.as_any().downcast_ref::<Insert>().unwrap();
-                    if insert_into.table.meta.is_temp {
-                        self.set(Self::InsertIntoTempTable(insert_into.table.clone()))?;
+                    if insert_into.runtime_preference == RuntimePreference::Local {
+                        let provider = match &insert_into.provider {
+                            ProviderReference::Provider(provider) => provider.clone(),
+                            ProviderReference::RemoteReference(_) => unreachable!(
+                                "required local table, found remote reference to table"
+                            ),
```

Comment:
Should I return an internal error instead of `unreachable!` here?

Reply:
I think this would be considered a programmer bug. If that's the case, then the panic seems reasonable to me.